### ISSUE TITLE
Cannot create new version of an API if old version has a scope assigned to a resource

### ISSUE THREAD

**Issue by chamilaadhi** [wso2/product-apim#2978]
**Description:**

Cannot create new version of an API if old version has a scope assigned to a resource

following error is thrown

`[2018-03-10 18:40:22,105] ERROR - UserAwareAPIProvider Failed to create new version : 2.0 of : PizzaShackAPI
org.wso2.carbon.apimgt.api.APIManagementException: Scope 'admin' already exists.
	at org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO.addScopes(ApiMgtDAO.java:7163)
	at org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO.addAPI(ApiMgtDAO.java:4878)
	at org.wso2.carbon.apimgt.impl.APIProviderImpl.createNewAPIVersion(APIProviderImpl.java:2199)
	at org.wso2.carbon.apimgt.impl.UserAwareAPIProvider.createNewAPIVersion(UserAwareAPIProvider.java:68)
	at org.wso2.carbon.apimgt.hostobjects.APIProviderHostObject.jsFunction_createNewAPIVersion(APIProviderHostObject.java:3515)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mozilla.javascript.MemberBox.invoke(MemberBox.java:126)
	at org.mozilla.javascript.FunctionObject.call(FunctionObject.java:386)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.jaggeryjs.rhino.publisher.modules.api.c6._c_anonymous_1(/publisher/modules/api/copy.jag:8)
	at org.jaggeryjs.rhino.publisher.modules.api.c6.call(/publisher/modules/api/copy.jag)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2430)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:269)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:97)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.jaggeryjs.rhino.publisher.modules.api.c0._c_anonymous_6(/publisher/modules/api/module.jag:22)
	at org.jaggeryjs.rhino.publisher.modules.api.c0.call(/publisher/modules/api/module.jag)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.jaggeryjs.rhino.publisher.site.blocks.overview.ajax.c0._c_anonymous_1(/publisher/site/blocks/overview/ajax/overview.jag:28)
	at org.jaggeryjs.rhino.publisher.site.blocks.overview.ajax.c0.call(/publisher/site/blocks/overview/ajax/overview.jag)
	at org.mozilla.javascript.optimizer.OptRuntime.call0(OptRuntime.java:23)
`

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

1. Create an API
2. Assign scope to resource
3. Create a new version of that API

**Related Issues:**

**Comment by chamilaadhi** [comment]
duplicate https://github.com/wso2/product-apim/issues/2987

**Comment by bhathiya** [comment]
Other part of the fix: https://github.com/wso2/carbon-apimgt/pull/5253
